posthog/tasks/calculate_event_property_usage.py

```diff
@@ -13,85 +14,140 @@
-
-    events_volume = _get_events_volume(team)
-    for event, value in event_names.items():
-        value["volume"] = _extract_count(events_volume, event)
-        event_names[event] = value
-
-    _save_team(team, event_names, event_properties)
-
-    properties_volume = _get_properties_volume(team)
-    for key, value in event_properties.items():
-        value["volume"] = _extract_count(properties_volume, key)
-        event_properties[key] = value
-
-    _save_team(team, event_names, event_properties)
-
-
-def _get_properties_volume(team: Team) -> List[Tuple[str, int]]:
-    timestamp = now() - timedelta(days=30)
-    if is_ee_enabled():
-        from ee.clickhouse.client import sync_execute
-        from ee.clickhouse.sql.events import GET_PROPERTIES_VOLUME
-
-        return sync_execute(GET_PROPERTIES_VOLUME, {"team_id": team.pk, "timestamp": timestamp},)
-    cursor = connection.cursor()
-    cursor.execute(
-        "SELECT json_build_array(jsonb_object_keys(properties)) ->> 0 as key1, count(1) FROM posthog_event WHERE team_id = %s AND timestamp > %s group by key1 order by count desc",
-        [team.pk, timestamp],
-    )
-    return cursor.fetchall()
-
-
-def _get_events_volume(team: Team) -> List[Tuple[str, int]]:
-    timestamp = now() - timedelta(days=30)
-    if is_ee_enabled():
+    CalculateEventPropertyUsage().run()
+
+
+class CalculateEventPropertyUsage:
+    _teams: QuerySet
+    _event_names_dict: Dict[int, Dict[str, Dict[str, Union[int, str]]]] = {}
+    _event_properties_dict: Dict[int, Dict[str, Dict[str, Union[int, str]]]] = {}
+    # number of events at which we query events individually
+    _clickhouse_volume_cutoff = 1000000
+
+    def __init__(self, clickhouse_volume_cutoff: Optional[int] = None) -> None:
+        # Purely set in tests
+        if clickhouse_volume_cutoff:
+            self._clickhouse_volume_cutoff = clickhouse_volume_cutoff
+
+    def run(self) -> None:
+        self._teams = Team.objects.all()
+        for team in self._teams:
+            self.calculate_usage_count_for_team(team=team)
+
+        self.calculate_event_volume()
+
+        if is_ee_enabled():
+            self._clickhouse_calculate_properties_volume()
+        else:
+            volume = self._psql_get_properties_volume()
+            self.save_properties_volume(volume)
+
+    def _save_team(self, team: Team) -> None:
+        def _sort(to_sort: List) -> List:
+            return sorted(to_sort, key=lambda item: (item.get("usage_count", 0), item.get("volume", 0)), reverse=True)
+
+        team.event_names_with_usage = _sort([val for _, val in self._event_names_dict[team.pk].items()])
+        team.event_properties_with_usage = _sort([val for _, val in self._event_properties_dict[team.pk].items()])
+        team.save()
+
+    def calculate_usage_count_for_team(self, team: Team) -> None:
+        self._event_names_dict[team.pk] = {event: {"event": event, "usage_count": 0} for event in team.event_names}
```

Comment:
Semantics question: Do we care about items which have usage count of 0? Asking to propose a refactoring.

Reply:
Yes we want those in this dict, as we use this dict to show the select box in the frontend